python/ray/experimental/state.py

```diff
@@ -345,10 +348,22 @@ def task_profiles(self):
         executions of that task. The second element is a list of profiling
         information for tasks where the events have no task ID.
     """
+    if start == None:
+      start = 0
+    if end == None:
+      end = time.time()
+    if num == None:
+      num = sys.maxsize
     task_info = dict()
     event_names = self.redis_client.keys("event_log*")
+    start_window = start
+    end_window = end
+    smallest_heap = []
```

Comment:
The comment should be pretty extensive, e.g., it should make it clear how the heap is being used, what the key is, whether the thing that gets popped off is the earliest or most recent event, etc.